checkpoint.go

```diff
@@ -36,7 +36,8 @@ type CheckParams struct {
 	Version string
 
 	// Generic product flags
-	Flags map[string]string
+	Flags      map[string]string
+	ExtraFlags func() []Flag
```

Comment:
For the record: @bboreham confirmed this is used by Scope at the time of writing.